test/newsimplification/Simplifier.spec.ts

```diff
@@ -0,0 +1,147 @@
+import {simplify, Simplifier} from '../../src/newsimplification/Simplifier';
+import {isModelObject, simplifyName} from '../../src/newsimplification/Utils';
+import simplifyProperties from '../../src/newsimplification/SimplifyProperties';
+import simplifyAllOf from '../../src/newsimplification/SimplifyAllOf';
+import { CommonModel, Schema } from '../../src/models';
+
+let mockedIsModelObjectReturn = false;
+jest.mock('../../src/newsimplification/Utils', () => {
+  return {
+    simplifyName: jest.fn(),
+    isModelObject: jest.fn().mockImplementation(() => {
+      return mockedIsModelObjectReturn;
+    })
+  }
+});
+jest.mock('../../src/newsimplification/SimplifyProperties');
+jest.mock('../../src/newsimplification/SimplifyAllOf');
+CommonModel.mergeCommonModels = jest.fn();
+/**
+ * Some of these test are purely theoretical and have little if any merit 
+ * on a JSON Schema which actually makes sense but are used to test the principles.
+ */
+describe('Simplification', function() {
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockedIsModelObjectReturn = false;
+  });
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+  test('should return empty models if false schema', function () {
+    const schema: any = false;
+    const models = simplify(schema);
+    expect(models).toHaveLength(0);
+  });
+  test('should return model with all types if true schema', function () {
+    const schema: any = true;
+    const models = simplify(schema);
+    expect(models).toHaveLength(1);
+    expect(models[0].type).toEqual(['object', 'string', 'number', 'array', 'boolean', 'null', 'integer']);
+  });
+  test('should set id of model if object', function() {
+    const schema = { type: 'object' };
+    const models = simplify(schema);
+    expect(models).toHaveLength(1);
+    expect(simplifyName).toHaveBeenNthCalledWith(1, schema);
```

Comment:
I have no idea from simplifyName is coming from. I do not like this.

Reply:
Utility function to realize the name of the schema (`src/newsimplification/Utils`).